src/services/Content/ContentQueries.ts

```diff
@@ -195,6 +261,62 @@ export const useValidateContentList = () => {
   });
 };
 
+export const useDeletePopularRepositoryMutate = (
+  queryClient: QueryClient,
+  page: number,
+  perPage: number,
+  filterData?: FilterData,
+) => {
+  const popularRepositoriesKeyArray = [
+    POPULAR_REPOSITORIES_LIST_KEY,
+    page,
+    perPage,
+    undefined,
+    ...Object.values(filterData || {}),
+  ];
+  const { notify } = useNotification();
+  return useMutation(deleteContentListItem, {
+    onMutate: async (uuid: string) => {
+      // Cancel any outgoing refetches (so they don't overwrite our optimistic update)
+      await queryClient.cancelQueries(popularRepositoriesKeyArray);
+      // Snapshot the previous value
+      const previousPopularData: Partial<PopularRepositoriesResponse> =
+        queryClient.getQueryData(popularRepositoriesKeyArray) || {};
+
+      queryClient.setQueryData(popularRepositoriesKeyArray, () => ({
+        ...previousPopularData,
+        data: previousPopularData.data?.map((data) => {
+          if (data.uuid === uuid) {
+            return { ...data, uuid: undefined };
+          }
+          return data;
```

Comment:
I'm probably not reading this correctly, but it looks like it optimistically updates the list to include the item but with a uuid of undefined?

Reply:
This is accurate. I guess I could use "", as that make !uuid evaluate to true.

Reply:
@Andrewgdewar Re the above comment, are we waiting for one more change?